lib/matplotlib/axes/_axes.py

```diff
@@ -6636,8 +6655,8 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
         if bin_range is not None:
             bin_range = self.convert_xunits(bin_range)
 
-        # Check whether bins or range are given explicitly.
-        binsgiven = np.iterable(bins) or bin_range is not None
+        # this in True for 1D arrays, and False for None and str
+        bins_array_given = np.ndim(bins) == 1
```

Comment:
At this point you may as well just put this inside `histogram_bin_edges`, since it's there in the non-backport